internal/loadbalancebenchmark/listfactory.go

```diff
@@ -0,0 +1,48 @@
+package loadbalancingbenchmark
+
+import (
+	"fmt"
+
+	"go.uber.org/multierr"
+	"go.uber.org/yarpc/api/peer"
+	"go.uber.org/yarpc/peer/pendingheap"
+	"go.uber.org/yarpc/peer/roundrobin"
+)
+
+type ListFactory func(group *ClientGroup) (peer.ChooserList, error)
+
+var listFactories = make(map[ListType]ListFactory)
+
+func NewRoundRobinList(group *ClientGroup) (peer.ChooserList, error) {
+	return roundrobin.New(NewBenchTransport()), nil
+}
+
+func NewFewestPendingList(group *ClientGroup) (peer.ChooserList, error) {
+	return pendingheap.New(NewBenchTransport()), nil
+}
+
+func RegisterList(balancingType ListType, factory ListFactory) error {
+	if factory == nil {
+		return fmt.Errorf(`unable to register %q, factory is nil`, balancingType)
+	}
+	if _, ok := listFactories[balancingType]; ok {
+		return fmt.Errorf(`factory for %q already exists`, balancingType)
+	}
+	listFactories[balancingType] = factory
+	return nil
+}
+
+func InitListFactory() error {
```

Comment:
What if we created a struct that holds this state and remove the globals? Since this file is in the same package as the rest of the implementation, files like `config.go` for instance still have access to `listFactories`.

Reply:
use a struct to wrap up these fields, make them unaccessible for other files.